src/Orleans/Lifecycle/ILifecycleObservable.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Observable lifecycle.
+    /// Each stage of lifecycle is observable.  All observers will be notified 
+    ///   when the stage is reached when starting, and stopping.
+    /// Stages are started in ascending order, and stopped in decending order.
+    /// </summary>
+    /// <typeparam name="TStage"></typeparam>
+    public interface ILifecycleObservable<in TStage>
```

Comment:
Maybe, but the problem is, that rules out primitive types, like ints and enums, which I think will be the most common types people would like to use.